src/Orc.WorkspaceManagement/Managers/WorkspaceManager.cs

```diff
@@ -119,6 +119,8 @@ public IWorkspace Workspace
 
             if (Directory.Exists(baseDirectory))
             {
+                _workspaces.Clear();
```

Comment:
I think the workspaces should be cleared anyway (even when directory doesn't exist). I think it's important to put the clear outside the if (even before the Initializing method)